### ISSUE TITLE
stdin should not be assumed to be an lvalue

### ISSUE THREAD

**Issue by pszkotak** [openthread/wpantund#89]
Hi Team,

wpanctl.c fails to compile with the following message when HAVE_LIBREADLINE is defined.

```
wpanctl.c: In function 'main':
wpanctl.c:485:10: error: assignment of read-only variable 'stdin'
    stdin = fopen(optarg, "r");
          ^
make[7]: *** [wpanctl-wpanctl.o] Error 1 
```

**Comment by darconeous** [comment]
What platform are you building on?

**Comment by pszkotak** [comment]
I'm cross-compiling on x86 with openwrt build system.
Target is the cortex A53, gcc 5.3.0 used, in case that matters.

**Comment by darconeous** [comment]
Ah. Well, it looks like they've made `stdin` constant on your platform. C89 does allow implementations to make `stdin` not be an lvalue, so this is indeed a bug.

[The fix for this is to use `freopen()`](http://stackoverflow.com/questions/584868/rerouting-stdin-and-stdout-from-c) instead of `fopen()`, which seems pretty straightforward.

**Comment by darconeous** [comment]
@pszkotak, Please give the proposed fix in #110 a try.
